apps/website/src/pages/index.tsx

```diff
@@ -140,7 +143,71 @@ const getTwitchEmbed = (
   return url.toString();
 };
 
-const Home: NextPage = () => {
+// Define the schema for the expected structure of the feed entries
+const VideoSchema = z.object({
+  videoId: z.string(),
+  title: z.string(),
+  published: z.date(),
+});
+
+type Video = z.infer<typeof VideoSchema>;
+
+// Define the schema for the entire feed
+const FeedSchema = z.object({
+  feed: z.object({
+    entry: z.array(
+      z.object({
+        "yt:videoId": z.array(z.string()).nonempty(),
+        title: z.array(z.string()).nonempty(),
+        published: z.array(z.string()).nonempty(),
+      }),
+    ),
+  }),
+});
+
+const fetchYouTubeFeed = async (channelId: string): Promise<Video[]> => {
+  const url = `https://www.youtube.com/feeds/videos.xml?channel_id=${channelId}`;
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch YouTube feed: ${response.statusText}`);
+  }
+
+  const xml = await response.text();
+  const json = await parseStringPromise(xml);
+  const parsedFeed = FeedSchema.parse(json);
+
+  return parsedFeed.feed.entry
+    .map((entry) => ({
+      videoId: entry["yt:videoId"][0],
+      title: entry.title[0],
+      published: new Date(entry.published[0]),
+    }))
+    .filter((video) => !video.title.includes("#shorts")) // exclude shorts
+    .map((video) => VideoSchema.parse(video));
+};
+
+export const getStaticProps: GetStaticProps = async () => {
+  const channelIds = ["UCbJ-1yM55NHrR1GS9hhPuvg", "UCfisf6HxiQr8_4mctNBm9cQ"];
+  const videosArrays = await Promise.all(
+    channelIds.map((channelId) => fetchYouTubeFeed(channelId)),
+  );
+
+  // Combine videos from all channels and sort by published date
+  const combinedVideos = videosArrays
+    .flat()
+    .sort((a, b) => b.published.getTime() - a.published.getTime());
+  const latestVideos = combinedVideos.slice(0, 4);
+
+  return {
+    props: {
+      videos: latestVideos,
+    },
+    revalidate: 1800, // revalidate after 30 minutes
+  };
+};
+
+const Home: NextPage<{ videos: Video[] }> = ({ videos }) => {
```

Comment:
Can we use `InferGetStaticPropsType` like we do on some other pages, so we just infer the props based on what `getStaticProps` returns, rather than hard-coding the prop types here?

Reply:
Take a look at the show and tell index page as a reference point :)

Reply:
my bad i completely misunderstood what you meant haha - fixed now